aim/web/ui_v2/src/utils/d3/drawHoverAttributes.ts

```diff
@@ -355,117 +377,93 @@ function drawHoverAttributes(props: IDrawHoverAttributesProps): void {
       .classed('active', false)
       .classed('focus', false);
 
-    attributesNodeRef.current.select('#HoverLine-x').remove();
-    if (yAxisLabelNodeRef.current) {
-      yAxisLabelNodeRef.current.remove();
-      yAxisLabelNodeRef.current = null;
-    }
+    clearHorizontalAxisLine();
+    clearYAxisLabel();
   }
 
-  function updateFocusedChart(
-    mousePos: [number, number],
+  function drawUpdateAttributes(
+    circle: INearestCircle,
+    nearestCircles: INearestCircle[],
     force: boolean = false,
   ): IActivePoint {
-    const { mouseX, mouseY } = getCoordinates({
-      mouse: mousePos,
-      margin,
-      xScale: attributesRef.current.xScale,
-      yScale: attributesRef.current.yScale,
-    });
-    const nearestCircles = getNearestCircles(mouseX);
-    const closestC = getClosestCircle(nearestCircles, mouseX, mouseY);
-
     // hover Line Changed case
-    if (force || closestC.key !== attributesRef.current.lineKey) {
+    if (force || circle.key !== attributesRef.current.lineKey) {
       linesNodeRef.current.classed(
         'highlight',
         highlightMode !== HighlightEnum.Off,
       );
       clearActiveLine(attributesRef.current.lineKey);
-      drawActiveLine(closestC.key);
+      drawActiveLine(circle.key);
     }
 
     // hover Circle Changed case
     if (
       force ||
-      closestC.key !== attributesRef.current.activePoint?.key ||
-      closestC.x !== attributesRef.current.activePoint?.xPos ||
-      closestC.y !== attributesRef.current.activePoint?.yPos
+      circle.key !== attributesRef.current.activePoint?.key ||
+      circle.x !== attributesRef.current.activePoint?.xPos ||
+      circle.y !== attributesRef.current.activePoint?.yPos
     ) {
       attributesNodeRef.current.classed(
         'highlight',
         highlightMode !== HighlightEnum.Off,
       );
       drawCircles(nearestCircles);
-      drawVerticalAxisLine(closestC.x);
-      drawHorizontalAxisLine(closestC.y);
-      drawXAxisLabel(closestC.x);
-      drawYAxisLabel(closestC.y);
-      drawActiveCircle(closestC.key);
+      drawVerticalAxisLine(circle.x);
+      drawHorizontalAxisLine(circle.y);
+      drawXAxisLabel(circle.x);
+      drawYAxisLabel(circle.y);
+      drawActiveCircle(circle.key);
     }
 
-    const activePoint = getActivePoint(closestC);
+    const activePoint = getActivePoint(circle);
     attributesRef.current.xStep = activePoint.xValue;
     attributesRef.current.activePoint = activePoint;
     return activePoint;
   }
 
+  function updateFocusedChart(
+    mousePos: [number, number],
+    force: boolean = false,
+  ): IActivePoint {
+    const { mouseX, mouseY } = getCoordinates({
+      mouse: mousePos,
+      margin,
+      xScale: attributesRef.current.xScale,
+      yScale: attributesRef.current.yScale,
+    });
+    const nearestCircles = getNearestCircles(mouseX);
+    const closestC = getClosestCircle(nearestCircles, mouseX, mouseY);
```

Comment:
@KaroMourad maybe it would be better to give this variable full name (`closestCircle`), what do you think?

Reply:
i think you right it would be more readable, i will fix it now